PyDSS/ProfileManager/ProfileStore.py

```diff
@@ -0,0 +1,150 @@
+
+        filePath = options['Profiles']["Profile store path"]
+        if os.path.exists(filePath):
+            self._logger.info("Loading existing h5 store")
+            self.store = h5py.File(filePath, mode)
+        else:
+            self._logger.info("Creating new h5 store")
+            self.store = h5py.File(filePath, "w")
+            for profileGroup in self.PROFILE_TYPES:
+                self.store.create_group(profileGroup)
+        return
+
+    def setup_profiles(self):
+        self.Profiles = {}
+        for group, profileMap in self.profileMapping.items():
+            if group in self.store:
+                grp = self.store[group]
+                for profileName, mappingDict in profileMap.items():
+                    if profileName in grp:
+                        objects = {x['object'] : self.Objects[x['object']] for x in mappingDict}
+                        self.Profiles[f"{group}/{profileName}"] = TSP(grp[profileName], objects, self.dssSolver,
+                                                                      mappingDict)
+                    else:
+                        self._logger.warning("Group {} \ data set {} not found in the h5 store".format(
+                            group, profileName
+                        ))
+            else:
+                self._logger.warning("Group {} not found in the h5 store".format(group))
+        return
+
+    def create_dataset(self, dname, pType, data ,startTime, resolution, units, info):
+        grp = self.store[pType]
+        if dname not in grp:
+            dset = grp.create_dataset(
+                dname,
+                data=data,
+                shape=(len(data),),
+                maxshape=(None,),
+                chunks=True,
+                compression="gzip",
+                compression_opts=4
+            )
+            self.createMetadata(
+                dset, startTime, resolution, data, units, info
+            )
+        else:
+            self._logger.warning('Dataset "{}" already exists in group "{}".'.format(dname, pType))
+
+    def add_from_arrays(self, data, name, pType, startTime, resolution, units="", info=""):
+        data = np.array(data)
+        r, c = data.shape
+        if r > c:
+            for i in range(c):
+                d = data[:, i]
+                dname = name if i==0 else "{}_{}".format(name, i)
+                self.create_dataset(dname=dname, pType=pType, data=d, startTime=startTime, resolution=resolution,
+                                    units=units, info=info)
+        else:
+            for i in range(r):
+                d = data[i, :]
+                dname = name if i==0 else "{}_{}".format(name, i)
+                self.create_dataset(dname=dname, pType=pType, data=d, startTime=startTime, resolution=resolution,
+                                    units=units, info=info)
+        return
+
+    def add_profiles_from_csv(self, csv_file, name, pType, startTime, resolution_sec=900, units="",
```

Comment:
I don't see that any of the add_* methods are called.  Am I missing something?

Reply:
I will be using it with the cli (yet to be built out).
I have already tested it. Will allow users to add existing profiles to theprofile store.